Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2243,44 +2244,78 @@ extension BrowserViewController: UIAdaptivePresentationControllerDelegate {
 extension BrowserViewController {
     func presentIntroViewController(_ alwaysShow: Bool = false) {
         if showLoadingScreen(for: .shared) {
-            present(LoadingScreen(profile: profile, referrals: referrals, referralCode: User.shared.referrals.pendingClaim), animated: true)
+            presentLoadingScreen()
         } else if User.shared.firstTime {
-            User.shared.firstTime = false
-            User.shared.migrated = true
-            User.shared.hideRebrandIntro()
-            // deactivate searchbar hint for new users
-            contextHintVC.viewModel.markContextualHintPresented()
+            handleFirstTimeUserActions()
         } else if User.shared.showsRebrandIntro {
-            let intro = NTPIntroViewController()
-            intro.modalPresentationStyle = .overFullScreen
-            intro.modalTransitionStyle = .crossDissolve
-            present(intro, animated: true)
-            User.shared.hideRebrandIntro()
+            presentRebrandIntro()
```

Comment:
I already mentioned this in Luca's PR. We can actually remove the rebrand intro and attached logic. If it eases your task, you could include it in this task, or we'll do it in an extra one

Reply:
My suggestion would be to approach the removal of this Rebrand Intro logic completely as a separate tasks so that we'll have a separate PR as well